### PR TITLE
feat: (PRO-82) add configuration validation enhancements and CLI options

### PR DESCRIPTION
**Pull request by dev-jodee**
- Introduced `validate_config` flag in CLI for optional configuration validation.
- Added `validate_with_result` method in `Config` for detailed validation results.
- Updated Makefile to streamline linting and formatting commands.
- Refactored validation logic to provide warnings and errors for configuration issues.
- Enhanced tests for validation scenarios, ensuring comprehensive coverage of edge cases.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,10 +27,11 @@ setup-test-env:
 	cargo run -p tests --bin setup-test-env
 
 # Run integration tests
-test-integration:
-	cargo run -p tests --bin setup-test-env
+test-integration: setup-test-env
 	cargo test --test integration
 
+test-all: test test-integration
```
Comment (amilz) [inline Makefile]: wdyt about adding client tests to this command?

File: crates/lib/src/config.rs [hunk 2]
```diff
@@ -164,6 +206,130 @@ impl Config {
...
+            warnings.push("Max signatures is 0 - this will block all transactions".to_string());
+        }
+
+        // Validate price source (warn if Mock)
+        if matches!(self.validation.price_source, PriceSource::Mock) {
+            warnings.push("Using Mock price source - not suitable for production".to_string());
+        }
+
+        // Validate allowed programs (warn if empty or missing system/token programs)
+        if self.validation.allowed_programs.is_empty() {
```
Comment (amilz) [inline crates/lib/src/config.rs:245]: wdyt about adding a `check_valid_program` to the program validation...valid address && owner = `BPFLoaderUpgradeab1e11111111111111111111111`

File: crates/lib/src/config.rs
Comment (amilz) [inline crates/lib/src/config.rs:1]: wdyt about fee_payer_policy warnings?

**Review by amilz — APPROVED** [review]
lgtm. couple suggestions

File: crates/lib/src/config.rs (hunk 2 above)
Comment (dev-jodee) [reply crates/lib/src/config.rs:245]: Yeah I thought about it, but I'm scared that validation using RPC will be kind of problematic, we could validate tokens, programs, etc. and check if signer, owner, etc. but RPCs aren't always stable and it could potentially spam the RPC. For now I prefer non rpc check but maybe if we get asked we could go with the full validation

File: crates/lib/src/config.rs
Comment (dev-jodee) [reply crates/lib/src/config.rs:1]: Yeah thought about it but honestly not 100% sure what the warning could be, like assign is disabled should we just flag it or explain what it disables, and if so then the warnings might become very annoying? What do you think

Comment (amilz) [reply crates/lib/src/config.rs:1]: fair. maybe at some point we could add a `verbose` flag or something if folks ask for more info. i think fine as-is for now.

File: crates/lib/src/config.rs (hunk 2 above)
Comment (amilz) [reply crates/lib/src/config.rs:245]: sgtm
